modules/@apostrophecms/doc/index.js

```diff
@@ -1093,6 +1097,67 @@ module.exports = {
           }
         });
       },
+
+      // Iterate through the document's field, and execute the provided handlers
+      // for each array, object and relationship field type.
+      // The handlers take the `doc` and the current `field` values as arguments,
+      // letting you edit them if needed.
+      //
+      // Note that the provided array and object handlers must call the `recursiveFunc`
+      // so that every nested fields are walked through.
+      // This this done by default, when array or object handlers are not provided.
+
+      walkThrough(
+        doc,
+        {
+          arrayHandler = (doc, field, recursiveFunc) => {
```

Comment:
I think writing an entire function inline as a default is not good for readability, please set it if the argument is null inside the method.

Reply:
I think the default handlers will wind up doing nothing because the recursion should be provided all the time after the handler is called.

Reply:
Normally they will be taken into account, as they should in `getRelatedDocsIds()`:

`walkThrough` is called with the `relationship` handler only, so the `array` and `object` handlers will be the default ones.

And in the case of a recursion (i.e area field type), `walkThrough` takes the provided handlers, or the default ones that are not provided.

Unless I'm missing something 🤔